lib/src/main/scala/spinal/lib/bus/localbus/sim/MemVIP.scala

```diff
@@ -0,0 +1,108 @@
+package spinal.lib.bus.localbus
+
+import spinal.core._
+import spinal.core.sim._
+class MemVIP{
+  private val mem = scala.collection.mutable.HashMap[Long, BigInt]()
+  def load(data:List[(Long, BigInt)]): Unit = mem ++= data
+  def add(addr: Long, data: BigInt): Unit = mem += addr -> data
+  def lookUp(addr: Long): BigInt = mem.getOrElse(addr, BigInt("FEFEFEFE", 16))
+  def loadData(data: Map[Long, BigInt]) = mem ++= data
+  def showMem = mem.map(t => s"${t._1.toHexString}_${t._2.toString(16)}").map(println)
+  def hang(bus: MemBus)(cd: ClockDomain): MemVIP = {
+    hangwrite(bus)(cd)
+    hangread(bus)(cd)
+    this
+  }
+
+  private def hangread(bus: MemBus)(cd: ClockDomain) ={
+    fork{
+      bus.rdat  #= 0
+      while(true){
+        cd.waitSampling()
+        if(bus.ce.toBoolean && !bus.wr.toBoolean ){
+          bus.rdat #= lookUp(bus.addr.toLong)
+        } else {
+          bus.rdat #= BigInt(0)
+        }
+      }
+    }
+  }
+
+  private def hangwrite(bus: MemBus)(cd: ClockDomain) = {
+    fork{
+      while(true){
+        cd.waitSampling()
+        if(bus.ce.toBoolean && bus.wr.toBoolean){
+          add(bus.addr.toLong, bus.wdat.toBigInt)
+        }
+      }
+    }
+  }
+
+  def hang(bus: MinBus, delay: Int = 10)(cd: ClockDomain): MemVIP = {
```

Comment:
what does `hang` mean - the only thing that comes to my mind is that it gets stuck which doesn't appear to be the case - looking at this and the one above shouldn't these just be `apply` methods?